lib/sidekiq_unique_jobs/client/middleware.rb

```diff
@@ -44,14 +46,14 @@ def storage_method
       def old_unique_for?
         connection do |conn|
           conn.watch(payload_hash)
-          pid = conn.get(payload_hash).to_i
-          if pid == 1 || (pid == 2 && item['at'])
+          pid = conn.get(payload_hash)
+          if pid && (pid != 'scheduled' || item['at'])
             conn.unwatch
             nil
           else
             conn.multi do
               if expires_at > 0
-                conn.setex(payload_hash, expires_at, item['jid'])
+                conn.setex(payload_hash, expires_at, item['at'] ? 'scheduled' : item['jid'])
```

Comment:
Or is the jid not at all interesting until it ends up in the real queue? I have been traveling for 60 hours and feel a little sluggish